src/main/host/thread_ptrace.c

```diff
@@ -372,13 +370,19 @@ static void _threadptrace_updateChildState(ThreadPtrace* thread, StopReason reas
 
 static void _threadptrace_nextChildState(ThreadPtrace* thread) {
     // Wait for child to stop.
+
     int wstatus;
     if (waitpid(thread->base.nativePid, &wstatus, 0) < 0) {
         error("waitpid: %s", g_strerror(errno));
         return;
     }
+
     StopReason reason = _getStopReason(wstatus);
     _threadptrace_updateChildState(thread, reason);
+
+    if (thread->childState == THREAD_PTRACE_CHILD_STATE_EXITED) {
+        thread->returnCode = wstatus;
```

Comment:
`WEXITSTATUS(wstatus)` (see `man waitpid`)

Reply:
So I'm not sure if the `WEXITSTATUS` macro will return the signal that terminated the process, or if you need to call `WTERMSIG` for that. Originally I just wanted to return all of the bits so I wasn't missing anything. I'll do a test to check, but if you see this in the meantime and know whether it does, please let me know.

Reply:
In general the full contents of `wstatus` is a bit opaque afaik. The documented API in waitpid says to use the macros to make any sense out of it.

Yeah, the exit code might not necessarily be the signal number when killed by an unhandled signal, but it should be non-zero, which should be good enough for our purposes.

OTOH on normal exit `wstatus` might still be non-zero, and we wouldn't want to say there was a non-zero exit status.

Reply:
I ran a test, and if the process is killed by a signal, `WIFEXITED(wstatus)` and `WEXITSTATUS(wstatus)` will both return 0. So if we aim to model something like bash, we should return `WEXITSTATUS(wstatus)` if the process exited normally, or `128+WTERMSIG(wstatus)` if the process exited by a signal.

Reply:
SGTM